include/watchdog.h

```diff
@@ -1,116 +1,206 @@
 /*
- * Copyright (c) 2015 Intel Corporation.
+ * Copyright (c) 2017 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @file
+ * @brief Public API for watchdog drivers.
+ */
+
+#if defined(CONFIG_WDT_LEGACY_API)
+
+/*
+ * This is the default, and will be the way until the new API below
+ * will be enforced everywhere.
+ */
+#include <watchdog_legacy.h>
+
+#else
+
 #ifndef _WDT_H_
 #define _WDT_H_
 
 #include <zephyr/types.h>
-#include <device.h>
 #include <misc/util.h>
+#include <device.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#define WDT_MODE		(BIT(1))
-#define WDT_MODE_OFFSET         (1)
-#define WDT_TIMEOUT_MASK        (0xF)
+/**
+ * @brief Enable calling application callback function on watchdog timeout.
+ */
+#define WDT_MODE_CALLBACK	BIT(0)
 
-enum wdt_mode {
-	WDT_MODE_RESET = 0,
-	WDT_MODE_INTERRUPT_RESET
+/**
+ * @brief Pause watchdog timer when CPU is in sleep state.
+ */
+#define WDT_PAUSE_IN_SLEEP	BIT(1)
+
+/**
+ * @brief Pause watchdog timer when CPU is halted by the debugger.
+ */
+#define WDT_PAUSE_HALTED_BY_DBG		BIT(2)
```

Comment:
Currently this API has only one field, namely 'options' which requires bit-field defines. But this may change in the future in case the API is extended. E.g. we still need to provide a function to determine the boot reason as requested by @lpereira. To avoid confusion I propose to proceed above defines with WDT_OPT_ to make clear they refer to the options field. It's a common practice and it will make it consistent with I2S API. I.e. WDT_PAUSE_HALTED_BY_DBG should become WDT_OPT_PAUSE_HALTED_BY_DBG